TypeCobol/Analysis/AnalyzerProvider.cs

```diff
@@ -10,103 +12,177 @@ namespace TypeCobol.Analysis
     /// <summary>
     /// IAnalyzerProvider default implementation. Can be used directly by adding activation delegates
     /// with the appropriate AddActivator method or by subclassing it.
+    /// It can also be used to allow composition of multiple providers into one.
+    /// This ensures that the creation of IAnalyzers won't interrupt the parsing.
     /// </summary>
-    public class AnalyzerProvider : IAnalyzerProvider
+    public class AnalyzerProviderWrapper : IAnalyzerProvider
     {
+        /// <summary>
+        /// Method used to log exceptions.
+        /// <remarks>Should not be used directly, use the accessor Logger instead.</remarks>
+        /// </summary>
+        [NotNull]
+        private Action<string> _logger = DoNothing;
+
+        /// <summary>
+        /// Method used to log exceptions.
+        /// </summary>
+        protected Action<string> Logger
+        {
+            get => _logger;
+            private set => _logger = value ?? DoNothing;
```

Comment:
- the setter should be protected to let inheritors have a public parameterless constructor and later set the logger
- could you add parentheses around the assigned value: `set => _logger = (value ?? DoNothing);` it is unclear in the current form.